### PR TITLE
 issue 83

### PR DESCRIPTION
**Pull request by AspiringDev51**
**To Test**
run rails console
Rails.application.load_tasks
Rake::Task['doc:sentencing_counties'].invoke

### CODE REVIEW

File: lib/tasks/one-off/doc_sentencing_counties.rake [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'open-uri'
+
+namespace :doc do
+  desc 'Pull in attorneys from OK Bar site'
```
Comment (sgelbart) [inline lib/tasks/one-off/doc_sentencing_counties.rake:4]: I think this is the wrong description

File: lib/tasks/one-off/doc_sentencing_counties.rake [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    
+    
+ 
+    sent_counties.each do |sent|
+        
+        
+
+        sentence = Doc::Sentence.where(sentencing_county: sent).in_batches
+        
+        if county_court.include?  sent  and sent != 'UNKNOWN COUNTY COURT'
```
Comment (sgelbart) [inline lib/tasks/one-off/doc_sentencing_counties.rake:19]: why not do:
```suggestion
        if sent.uppercase.include? 'COUNTY COURT' and sent != 'UNKNOWN COUNTY COURT'
```
I think then you can remove line 6-8?

File: spec/factories/doc_sentencing_counties.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :doc_sentencing_county do
+    name { 'MyString' }
```
Comment (sgelbart) [inline spec/factories/doc_sentencing_counties.rb:3]: are we using faker?

File: spec/services/importers/party_data_spec.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
   describe '#perform(oscn_id)' do
     it 'saves aliases' do
       path = 'spec/fixtures/party_example.html'
-      html = File.open(path).read
+      html = File.read(path)
```
Comment (sgelbart) [inline spec/services/importers/party_data_spec.rb:7]: did you mean to commit this?

File: lib/tasks/one-off/doc_sentencing_counties.rake [hunk 5]
```diff
@@ -0,0 +1,38 @@
+require 'open-uri'
+
+namespace :doc do
+  desc 'Pull in attorneys from OK Bar site'
+  task :sentencing_counties do
+    county_court = ActiveRecord::Base.connection.execute("SELECT DISTINCT sentencing_county from doc_sentences where sentencing_county ilike '%county court%' order by sentencing_county")
+
+    county_court = county_court.column_values(0).sort
+    sent_counties = ::Doc::Sentence.distinct.pluck(:sentencing_county).sort
```
Comment (sgelbart) [inline lib/tasks/one-off/doc_sentencing_counties.rake:9]:
```suggestion
    sent_counties = ::Doc::Sentence.select(:sentencing_county).distinct
```
I'd also consider renaming  sent_counties to `sentencing_county_names` (it's longer but clearer what you're working with...if it's an object vs a string)

Comment (sgelbart) [inline lib/tasks/one-off/doc_sentencing_counties.rake:9]: I'd consider renaming  sent_counties to `sentencing_county_names` (it's longer but clearer what you're working with...if it's an object vs a string)
and also in your loop renaming to `sentencing_county_name`

File: db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class FixDocSentencingCountiesId < ActiveRecord::Migration[6.0]
+  def change
+    rename_column :doc_sentences, :doc_sentencing_counties_id, :doc_sentencing_county_id
```
[the remark this answers is not in the document]
Comment (AspiringDev51) [reply db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb:3]: I'm a little confused, are these different migration files wrong or bad? I thought It was good to have different migration files instead of one big one? plus If I remember correctly  this migration file is new to the project since its dealing with a new table. are you saying I should revert all my migrations that I've done to fit it into one file?

File: spec/services/importers/party_data_spec.rb (hunk 4 above)
Comment (AspiringDev51) [reply spec/services/importers/party_data_spec.rb:7]: Again i think this is a rubocop thing that auto corrects whenever I run it. it seems to prefer the `File.read(path)` over  'File.open(path).read' .

File: db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (sgelbart) [reply db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb:3]: I'd have one migration file per commit per table you're working on

**Review by sgelbart — APPROVED** [review]

File: db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (sgelbart) [reply db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb:3]: Not a big deal though, just something nice for future to keep migrations a bit cleaner

File: db/migrate/20220726202412_create_doc_sentencing_counties.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class CreateDocSentencingCounties < ActiveRecord::Migration[6.0]
+  def change
+    create_table :doc_sentencing_counties do |t|
+      t.string :name
```
Comment (holdenmitchell) [inline db/migrate/20220726202412_create_doc_sentencing_counties.rb:4]: I think it makes sense to match the backend constraint to the model constraint

```suggestion
      t.string :name, null: false
```

File: db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (holdenmitchell) [reply db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb:3]: ☝️ I agree with @sgelbart. We can just rollback and make this change on the original migration. We can pair on it on Friday

File: db/migrate/20220729190628_remove_null_restriction.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class RemoveNullRestriction < ActiveRecord::Migration[6.0]
+  def change
+    change_column_null :doc_sentencing_counties, :county_id, true
```
Comment (holdenmitchell) [inline db/migrate/20220729190628_remove_null_restriction.rb:3]: Same as comment above. Let's roll these into the original migration

File: db/migrate/20220729201159_add_null_restriction.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class AddNullRestriction < ActiveRecord::Migration[6.0]
+  def change
+    change_column_null :doc_sentencing_counties, :name, false
```
Comment (holdenmitchell) [inline db/migrate/20220729201159_add_null_restriction.rb:3]: Same as comment above. Let's roll these into the original migration

File: spec/models/doc_sentencing_county_spec.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe DocSentencingCounty, type: :model do
+  describe 'validations' do
```
Comment (holdenmitchell) [inline spec/models/doc_sentencing_county_spec.rb:4]: Don't forget to add in the association check for `belongs_to :county, optional: true`

File: db/schema.rb [hunk 11]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2022_05_03_152707) do
+ActiveRecord::Schema.define(version: 2022_07_29_201159) do
```
Comment (holdenmitchell) [inline db/schema.rb:13]: Lot's of unrelated schema changes. We can fix these when we pair

File: lib/tasks/one-off/doc_sentencing_counties.rake [hunk 12]
```diff
@@ -0,0 +1,23 @@
+require 'open-uri'
+
+namespace :doc do
+  desc 'Populates DocSentencingCounty and assignes it to Doc::Sentence'
+  task :sentencing_counties do
+    sent_counties.each do |sent|
```
Comment (holdenmitchell) [inline lib/tasks/one-off/doc_sentencing_counties.rake:6]: Having an issue with this line. Where does the sent_counties come from?

**Review by holdenmitchell — CHANGES_REQUESTED** [review]
I think we can go ahead and make some updates to the migrations as @sgelbart suggested. Also, there are some schema changes that we should resolve.

When I run the rake task, I get the following error. I commented on the problem link in the `doc_sentencing_counties.rake` file

```
NameError: undefined local variable or method `sent_counties' for main:Object
```

File: db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (AspiringDev51) [reply db/migrate/20220728193602_fix_doc_sentencing_counties_id.rb:3]: Sounds good!

File: db/migrate/20220726202412_create_doc_sentencing_counties.rb (hunk 7 above)
Comment (AspiringDev51) [reply db/migrate/20220726202412_create_doc_sentencing_counties.rb:4]: should I rollback and edit  the file here or make a new db:migrate? either way I assume we will pair on this on friday correct?

Comment (AspiringDev51) [reply db/migrate/20220726202412_create_doc_sentencing_counties.rb:4]: Also Fixed the error  by adding sent_counties back in,  I think I accidentally deleted it beforehand.

File: lib/tasks/one-off/doc_sentencing_counties.rake (hunk 12 above)
Comment (AspiringDev51) [reply lib/tasks/one-off/doc_sentencing_counties.rake:7]: I believe sent_counties was accidentally deleted. added it back in

File: spec/models/doc_sentencing_county_spec.rb (hunk 10 above)
Comment (AspiringDev51) [reply spec/models/doc_sentencing_county_spec.rb:4]: added in!

**Review by holdenmitchell — APPROVED** [review]
